corehq/apps/domain/decorators.py

```diff
@@ -315,13 +315,12 @@ def _inner(request, domain, *args, **kwargs):
 
 
 def _two_factor_required(view_func, domain, couch_user):
-    if TWO_FACTOR_SUPERUSER_ROLLOUT.enabled(couch_user.username):
-        return not getattr(view_func, 'two_factor_exempt', False)
-    return (
-        not getattr(view_func, 'two_factor_exempt', False)
-        and domain.two_factor_auth
-        and not couch_user.two_factor_disabled
-    )
+    if domain.two_factor_auth or (settings.ENFORCE_TWO_FACTOR_FOR_SUPERUSERS and couch_user.is_superuser):
+        return (
+            not getattr(view_func, 'two_factor_exempt', False)
```

Comment:
Looks like we have two different ways to bypass two factor. This and https://github.com/dimagi/commcare-hq/pull/20738/files#diff-e41763d8d0b6724fd2a313c6f2caa85eR503 @nickpell could you follow up and see if your recent PR could work with `two_factor_exempt` ?

Reply:
I did look at @nickpell PR but his approach was necessary since he just needed to make the one request exempt (which is a faked request) and not all requests. I think the smell here is related to how we do that 'faked' request in the reporting code.

Reply:
Agreed that the core smell is the fake request, but don't feel like that kind of refactor would fit into interrupt. thanks for looking into that to see that we did need it